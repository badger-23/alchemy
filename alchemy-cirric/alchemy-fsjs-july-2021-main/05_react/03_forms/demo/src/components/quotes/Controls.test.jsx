import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('Controls', () => {
  it('displays a form with an input and submit button', () => {
    render(
      <Controls
        characterName="Bender"
        onCharacterNameChange={() => {}}
        onSubmit={() => {}}
      />
    );

    const controlsForm = screen.getByTestId('controls-form');
    expect(controlsForm).toMatchSnapshot();
  });
});
