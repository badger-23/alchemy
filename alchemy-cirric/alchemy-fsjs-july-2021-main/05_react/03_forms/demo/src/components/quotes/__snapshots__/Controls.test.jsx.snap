// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls displays a form with an input and submit button 1`] = `
<form
  data-testid="controls-form"
>
  <label
    for="characterName"
  >
    Character Name
  </label>
  <input
    id="characterName"
    type="text"
    value="Bender"
  />
  <button
    aria-label="find-quotes"
  >
    Submit
  </button>
</form>
`;
